### ISSUE TITLE
损失函数公式

### ISSUE THREAD

**Issue by FolSpark** [yongliang-wu/DFT#3]
作者您好，从公式的角度理解：

交叉熵的公式：

$\mathcal{L}(\theta) = -\log\left(\pi(\theta)\right)$

其梯度为：

$\nabla_\theta \mathcal{L}(\theta) = -\frac{\nabla_\theta \pi(\theta)}{\pi(\theta)}$

您是在交叉熵的梯度中乘了一个概率的系数（因为停止梯度，这个系数可以理解为是直接乘到梯度上），这样梯度就是

$\nabla_\theta \mathcal{L}(\theta) = -\frac{\nabla_\theta \pi(\theta)}{\pi(\theta)}\cdot \pi(\theta)=-\nabla_\theta \pi(\theta)$

这样简化后，其实您这样修改后，就是不用交叉熵作为损失函数了，直接把概率加负数作为损失函数更好，是这个意思吗？

**Comment by yongliang-wu** [comment]
看起来在数学上是等价的，但是用 logits 的话数值稳定性更好，直接用 softmax 可能会溢出，我回头会尝试一下直接优化概率函数试试～
